File: JAVA/136_SingleNumber.java
// Name:- Siddhartha Mishra 
// Date:- 20/10/22
/*
  Leetcode problem:- 136 (Single Number)

 * Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.

You must implement a solution with a linear runtime complexity and use only constant extra space.

 

Example 1:

Input: nums = [2,2,1]
Output: 1
Example 2:

Input: nums = [4,1,2,1,2]
Output: 4
Example 3:

Input: nums = [1]
Output: 1
 

Constraints:

1 <= nums.length <= 3 * 104
-3 * 104 <= nums[i] <= 3 * 104
Each element in the array appears twice except for one element which appears only once.
 */

class Solution {
    public int singleNumber(int[] nums) 
    {
        int res = 0;
        for (int num : nums)
        {
            res ^= num;
        }
        
        return res; // Return the single Number from the array 
    }
}

File: C++/136_SingleNumber.cpp
// Name: Shradha Verma
// Date: 22 October 2022

// Used Bitwise XOR Operator 
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int result = 0;

        for(int i = 0; i < nums.size(); i++){   // Traverse through the vector and find the XOR of all the elements

            result ^= nums[i];      // Every element that appears twice will give XOR as zero.

        }
        return result;      // XOR of Zero and the element occuring once will result in that particular elemnet.
    }
};